### PR TITLE
feat: kubernetes request timeout configurable

### PR DESCRIPTION
**Pull request by mclarke47**
Part of https://github.com/backstage/backstage/issues/18510

Need to add docs

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes-backend/src/service/KubernetesFetcher.ts [hunk 1]
```diff
@@ -231,6 +234,7 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
         'Content-Type': 'application/json',
         Authorization: `Bearer ${clusterDetails.serviceAccountToken}`,
       },
+      timeout: this.requestTimeout,
```
[the remark this answers is not in the document]
Comment (mclarke47) [reply plugins/kubernetes-backend/src/service/KubernetesFetcher.ts:237]: Strange I can see it on the installed version
Screenshot 2023-07-27 at 2 30 27 PM

[the remark this answers is not in the document]
Comment (jamieklassen) [reply plugins/kubernetes-backend/src/service/KubernetesFetcher.ts:237]: Ah I see now, I was looking at the latest docs which are for v3.x, in which the timeout option was removed. We're using 2.6.7, and if I look at [those docs](https://www.npmjs.com/package/node-fetch/v/2.6.7#options) I do see the option there. Sorry about that!

[the remark this answers is not in the document]
Comment (jamieklassen) [reply plugins/kubernetes-backend/src/service/KubernetesFetcher.ts:237]: Can't find a link for it, but I know there has been some interest in migrating all the backend plugins to use the global `fetch` API available in Node 18+, to have a smooth resolution to stuff like [running your backend behind a corporate proxy](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/help-im-behind-a-corporate-proxy.md). If and when that happens, the global `fetch` doesn't have a `timeout` option so we'd have to refactor this. I think that's a level of tech debt I'm comfortable with right now.

[the remark this answers is not in the document]
Comment (jamieklassen) [reply plugins/kubernetes-backend/src/service/KubernetesFetcher.ts:237]: Looking in the [`node-fetch` source code](https://github.com/node-fetch/node-fetch/blob/v2.6.7/src/body.js) I still don't see a default value for the timeout. Is it possible that this 10s timeout is coming from somewhere else, maybe something in express?

**Comment by mclarke47** [comment]
I'm actually not sure this is an issue now, I'm trying to work out where the 504 is coming from

**Comment by mclarke47** [comment]
Closing until we figure out where the issue is

**Pull request closed without merging**